### PR TITLE
Makes cyborg ratvar conversion work properly

### LINKED ISSUE #18255: Cyborg servants of Ratvar not getting the law datum and antag hud
[Note]: # (State what the issue is from a "whats wrong" prospective. Issue reports should clearly allow maintainers to understand whats wrong and how to test/reproduce if that is not obvious. Avoid ambiguity. Start your issue report below both of these lines (or remove them))

[Admins]: # (If you are reporting a bug that occured AFTER you used varedit/admin buttons to alter an object out of normal operating conditions, please verify that you can re-create the bug without the varedit usage/admin buttons before reporting the issue.)
Cyborgs produced via shoving a soul vessel in a cyborg body don't get the antag hud or the law datum, though they're still servants of ratvar and in the check antag menu.

### LINKED ISSUE #18377: Clockwork cult "cog" antag HUD icon is not visible
Not visible for cultists or admins.

### PR DESCRIPTION
**Pull request by ChangelingRain**
Fixes #18255
Fixes #18377 
Incidental fixes/changes:
Mode converts servants with the proper proc, so they get the HUD.
Can't convert the chaplain, because they're the chaplain, can't convert the captain, same reason.
Can't convert a guardian unless its owner is converted.
Can't put a clockwork MMI in an AI core, as the fucking thing lacks support for MMI-based laws.
Instead of trying to convert a borg if the MMI was clockwork, it'll just update the HUD icons.
Laws for ratvar'd borgs are updated with the proper procs.
Admin servant adding/removal is silent.

### CODE REVIEW

**Comment by ChangelingRain** [comment]
aaaaaa gamemode and conversion code coiax is gonna murder me

**Comment by PKPenguin321** [comment]
>can't convert the captain

Lame

File: code/game/objects/structures/ai_core.dm [hunk 1]
```diff
@@ -144,7 +144,7 @@
 					user << "<span class='warning'>This MMI does not seem to fit!</span>"
 					return
 
-				if(M.hacked)
+				if(M.hacked || M.clockwork)
```
Comment (ChangelingRain) [inline code/game/objects/structures/ai_core.dm]: this is butts and I hate it but it's A technically a balance thing and B i don't feel like refactoring all this code to work with mmi laws right now

File: code/game/gamemodes/clock_cult/clock_cult.dm [hunk 2]
```diff
@@ -82,18 +101,20 @@ This file's folder contains:
 			R.emagged = 1
 			R << "<span class='warning'><b>You have been desynced from your master AI. In addition, your onboard camera is no longer active and your safeties have been disabled.</b></span>"
 		S.laws = new/datum/ai_laws/ratvar
+		S.laws.associate(S)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_cult.dm]: Very important; means you can't simply reset the AI or borg to remove the law, which would accordingly be very bad, as they'd still be an antag and it'd get confusing.

File: code/game/objects/items/robot/robot_parts.dm [hunk 3]
```diff
@@ -237,9 +240,10 @@
 				O.notify_ai(1)
 				if(forced_ai)
 					O.connected_ai = forced_ai
-			if(!lawsync && !M.hacked)
+			if(!lawsync)
 				O.lawupdate = 0
-				O.make_laws()
+				if(!M.hacked && !M.clockwork)
```
Comment (ChangelingRain) [inline code/game/objects/items/robot/robot_parts.dm]: Also quite important. We don't want to override the laws we just got and we don't want to sync.

File: code/game/gamemodes/clock_cult/clock_cult.dm [hunk 4]
```diff
@@ -106,7 +127,7 @@ This file's folder contains:
 			var/mob/living/silicon/robot/R = S
 			R.emagged = initial(R.emagged)
 			R << "<span class='warning'>Despite your freedom from Ratvar's influence, you are still irreparably damaged and no longer possess certain functions such as AI linking.</span>"
-		S.laws = initial(S.laws)
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_cult.dm]: i'm pretty sure the initial of this var is null

File: code/game/objects/structures/ai_core.dm (hunk 1 above)
Comment (GunHog) [inline code/game/objects/structures/ai_core.dm]: What if you could set the MMI to run an antag conversion proc on the AI's core before being deleted?

Comment (ChangelingRain) [inline code/game/objects/structures/ai_core.dm]: I mean, I could, but I don't want to fuck with this code right now so hey, stopgap fixes

**Pull request merged**